File: javase/src/cn/tedu/day07/object/ObjectDemo2.java
package cn.tedu.day07.object;

public class ObjectDemo2 {
    public static void main(String[] args) {
        //创建Baby类对象
        Baby b=new Baby();
        System.out.println(b.name+","+b.age);
    }
}


//代表婴儿的类
class Baby{
    //属性
    String name;
    int age;
    char gender;

    //构造代码块
    //可以对属进行初始化
    //在调用任意形式的构造方法时都会先执行构造代码块
    {
        this.name="lili";
        this.age=1;
        System.out.println("构造代码块");
    }

    //无参构造
    public Baby(){
        /*this.name="lili";
        this.age=1;*/
        System.out.println("无参构造");
    }

    //有参构造
    public Baby(String name){
        //属性初始化
        //this.name=name;
//        this.name="lili";
//        this.age=1;
    }

    //有参构造
    public Baby(String name,int age){
       /* this.name=name;
        this.age=age;*/
//        this.name="lili";
//        this.age=1;
    }

    //方法---行为
    public void cry(){
        System.out.println(name+"就在嗷嗷嗷的哭...");
    }
}

File: J_scala/src/cn/gl/Demo/objectx/ObjectDemo2.scala
package cn.gl.Demo.objectx

object ObjectDemo2 {

  def main(args: Array[String]): Unit = {

    var d = new Driver("Bob", 19, "male");

  }

}

// 在构造器中定义的参数默认就是当前类的属性
// 主构造器
class Driver(name: String, age: Int, gender: String) {

  var no: String = _
  var addr: String = _

  // def setName(name:String) = this.name = name

  // 辅构造器
  def this(name: String, age: Int, gender: String, addr: String) {
    this(name, age, gender)
    this.addr = addr
  }

  // 辅构造器中必须先调用主构造器
  def this(name: String, gender: String) {
    this(name, 18, gender)
  }

}
